### PR TITLE
fix: master cannot download s3 from us-east-1 

### PR DESCRIPTION
**Pull request by NicholasBlaskey**
## Description

Bug in how we called the s3 API meaning we couldn't download from ```us-east-1```

## Test Plan

Integration test should cover it, manual though

manual

run ```mnist_pytorch_const``` with config
```
name: mnist_pytorch_const
hyperparameters:
  learning_rate: 1.0
  n_filters1: 32
  n_filters2: 64
  dropout1: 0.25
  dropout2: 0.5
searcher:
  name: single
  metric: validation_loss
  max_length:
      epochs: 1
  smaller_is_better: true
entrypoint: python3 train.py
checkpoint_storage:
  type: s3
  bucket: storage-unit-tests-us-east-1
  prefix: master/checkpoint-download
  access_key: EDIT
  secret_key: EDIT
```

direct download a checkpoint through ```det checkpoint download $CHECKPOINT_ID --mode master``` $CHECKPOINT_ID you can get from ```det e lc $expID```

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by tara-det-ai — APPROVED** [review]
LGTM

File: master/pkg/checkpoints/s3/s3checkpoints.go [hunk 1]
```diff
@@ -35,21 +36,27 @@ func (w *seqWriterAt) WriteAt(p []byte, off int64) (int, error) {
 // GetS3BucketRegion returns the region name of the specified bucket.
 // It does so by making an API call to AWS.
 func GetS3BucketRegion(ctx context.Context, bucket string) (string, error) {
-	sess, err := session.NewSession(&aws.Config{
-		Region: aws.String("us-west-2"),
-	})
+	// TODO this won't work on non aws s3 APIs.
```
Comment (stoksc) [inline master/pkg/checkpoints/s3/s3checkpoints.go:39]: unfortunately, if this isn't going to work within minio, it is sort of a non-starter. though, did this work before with minio, anyway?

can we have add an optional region field to `checkpoint_storage.type == s3` instead? trying to think about alternatives.

Comment (NicholasBlaskey) [reply master/pkg/checkpoints/s3/s3checkpoints.go:39]: I don't know how it could have worked with minio, we don't have the ```endpoint_url``` here

The caller can check ```endpoint_url``` to check if it is a non aws bucket. 

https://github.com/minio/minio/discussions/15063 minio seems to want us-east-1 so maybe we just default to that if ```endpoint_url``` is not empty and not s3?

Comment (NicholasBlaskey) [reply master/pkg/checkpoints/s3/s3checkpoints.go:39]: though I do think this is pretty separate from this change and something we would also want to test in CI

**Review by stoksc — APPROVED** [review]
lgtm, maybe have @hanyucui take a look too

File: master/pkg/checkpoints/s3/s3checkpoints.go (hunk 1 above)
Comment (NicholasBlaskey) [reply master/pkg/checkpoints/s3/s3checkpoints.go:39]: created https://hpe-aiatscale.atlassian.net/browse/DET-10035

do you know if minio is the only s3 clone people use? Or if there are others we should probably test with and make sure they work?
